test/signup.js

```diff
@@ -77,4 +79,37 @@ export async function verifyUserOverApi(username = 'automated@test.mail', passwo
     assert.equal(user.email, testUserEmail);
     assert.equal(user.first_name, testUserFirstName);
     assert.equal(user.last_name, testUserLastName);
+} */
+
+// This is a replacement of the above function `verifyUserOverApi` which requires the access token.
+// Note: access token can only requested after being loged in.
+export async function verifyUserByLogin(driver) {
+    await login(driver, testUserEmail, testUserPassword, verificationPageTitle);
+}
+
+export async function navigateToProfile(driver) {
+    await driver.get(`${url}/profile/profile`);
+
+    let title = await driver.getTitle();
+    assert.equal(title, 'Profile - Open Educational Badges');
+}
+
+/**
+ * This assumes that the driver already navigated to the profile page
+ */
+export async function deleteUserViaUI(driver) {
+    const menuButton = await driver.findElement(By.id(
+        'trigger2'));
+    await menuButton.click();
+
+    const dropdownButtons = await driver.findElements(By.id(
+        'menu2'));
+    const deleteButton = dropdownButtons[0];
+    await deleteButton.click();
+
+    const confirmDeleteButton = await driver.wait(until.elementLocated((By.xpath("//button[text()=' Account löschen ']"))), defaultWait)    
```

Comment:
I think it's nice to be consistent with how we retrieve elements; so far we've never used `xpath`. If you need a (confirm) button with a certain text, there's a helper for that in `util/selection.js` (called `submitButtonWithText`)

Reply:
atm the button in UI is still the old style. I added a comment to use `submitButtonWithText` after updating the UI.